examples_tests/35.TAA/taa.frag

```diff
@@ -48,10 +77,11 @@ void main()
 	vec3 aabb_min = min(current, min(neigh0, min(neigh1, min(neigh2, neigh3))));
 	vec3 aabb_max = max(current, max(neigh0, max(neigh1, max(neigh2, neigh3))));
 
-	history = clamp(history, aabb_min, aabb_max);
+	history = constrainHistSample(history, aabb_min, aabb_max);
 	
 	vec3 current_unjittered = textureLod(uCurrentBuf, TexCoords-uJitter, 0.0).rgb;
 	
+	// TODO use luma to vary interpolation factor as playdeadgames does
```

Comment:
nice idea.

Reply:
Also page 17 from Tiago Sousa had a funky idea, but that requires velocity info for both current and previous frame (so current matrix, 1 frame behind, and 2 frames behind).
https://www.slideshare.net/TiagoAlexSousa/graphics-gems-from-cryengine-3-siggraph-2013

And obviously to get v_p, you'd need to reproject pixel to previous frame, and then reproject again, then the diff between the reprojections would be v_p

Reply:
Actually luma weight is used by UE4 + INSIDE did their research and Sousa velocity weight didn't work for them that well... but you can try it once you have the basic TAA working to see whats best.

Reply:
Ok Uncharted and Lumberyard car game
https://aws.amazon.com/blogs/gametech/anti-ghosting-with-temporal-anti-aliasing/

Used stencil buffer to augment luma weight and color clipping (which should be done in YUV or YCoCg), basically if stencil fails then no history sample to be taken (weight = 0.0).

something similar could be done not with the stencil buffer, but with ObjectID buffer, in the far future.

Reply:
Also for reference
https://youtu.be/WzpLWzGvFK4?t=623